src/main/java/org/opensearch/flowframework/common/DefaultUseCases.java

```diff
@@ -171,4 +214,22 @@ public static String getSubstitutionReadyFileByUseCaseName(String useCaseName) t
         logger.error("Unable to find substitution ready file for use case: {}", useCaseName);
         throw new FlowFrameworkException("Unable to find substitution ready file for use case: " + useCaseName, RestStatus.BAD_REQUEST);
     }
+
+    /**
+     * Gets the required parameters based on the given use case
+     * @param useCaseName name of the given use case
+     * @return the list of required params
+     * @throws FlowFrameworkException if the use case doesn't exist in enum
+     */
+    public static List<String> getRequiredParamsByUseCaseName(String useCaseName) throws FlowFrameworkException {
+        if (useCaseName != null && !useCaseName.isEmpty()) {
+            for (DefaultUseCases useCase : values()) {
+                if (useCase.getUseCaseName().equals(useCaseName)) {
+                    return new ArrayList<String>(useCase.getRequiredParams());
+                }
+            }
+        }
+        logger.error("Unable to find required parameters for use case: {}", useCaseName);
+        throw new FlowFrameworkException("Unable to find required parameters for use case: " + useCaseName, RestStatus.BAD_REQUEST);
```

Comment:
This may be accurate but doesn't match the root cause of the error which is a bad use case, which is what the user needs to correct with the BAD_REQUEST.

Reply:
Sure I'll improve the error message